src/koheesio/spark/delta.py

```diff
@@ -300,7 +299,9 @@ def exists(self) -> bool:
         result = False
 
         try:
-            self.spark.table(self.table_name)
+            # In Spark remote session it is not enough to call just spark.table(self.table_name)
+            # as it will not raise an exception, we have to make action call on table to check if it exists
+            self.spark.table(self.table_name).take(1)
```

Comment:
can we run self.spark.table(self.table_name).take(1) only for remote session.

This is expensive operation for just to check if table exists or not

Reply:
Fixed